.pre-commit-config.yaml

```diff
@@ -104,8 +104,12 @@ repos:
               src/lightning_fabric/CHANGELOG.md
           )$
 
-  - repo: https://github.com/PyCQA/flake8
-    rev: 6.0.0
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: 'v0.0.226'
     hooks:
-      - id: flake8
-        name: Check PEP8
+      - id: ruff
+        # Respect `exclude` and `extend-exclude` settings.
+        args:
+          - "--fix"
+          - "--respect-gitignore"
+          - "--force-exclude"
```

Comment:
Why do we need this one?

```suggestion
```

Reply:
it is mentioned in ruff docs/example to follow wxlude but it is now much more explained... we can test it without

Reply:
tested and this needs to stay